src/mrpro/utils/filters.py

```diff
@@ -64,13 +70,12 @@ def filter_separable(
         # padding is done with pad() before the convolution
         padding_conv = 'valid'
 
+    # output will be of the promoted type of the input and the kernels
+    target_dtype = reduce(torch.promote_types, [k.dtype for k in kernels], x.dtype)
+    x = x.to(target_dtype)
+
     for kernel, ax in zip(kernels, axis, strict=False):
```

Comment:
did we agree to use `dim `instead of `axis `at some point? 
Although, I understand it is not relevant for that PR, maybe include this to some existing issue?

Reply:
I changed it for the filter functions